cli/azd/pkg/environment/environment.go

```diff
@@ -269,11 +269,28 @@ func (e *Environment) Save() error {
 		return fmt.Errorf("failed to create a directory: %w", err)
 	}
 
-	err = godotenv.Write(e.dotenv, filepath.Join(e.Root, azdcontext.DotEnvFileName))
+	// Instead of calling `godotenv.Write` directly, we need to save the file ourselves, so we can fixup any numeric values
+	// that were incorrectly unquoted.
+	marshalled, err := godotenv.Marshal(e.dotenv)
 	if err != nil {
 		return fmt.Errorf("saving .env: %w", err)
 	}
 
+	marshalled = fixupUnquotedDotenv(e.dotenv, marshalled)
+
+	envFile, err := os.Create(filepath.Join(e.Root, azdcontext.DotEnvFileName))
+	if err != nil {
+		return fmt.Errorf("saving .env: %w", err)
+	}
+	defer envFile.Close()
+
+	// Write the contents (with a trailing newline, and sync the file, as godotenv.Write)
+	if _, err := envFile.WriteString(marshalled + "\n"); err != nil {
+		return fmt.Errorf("saving .env: %w", err)
+	}
+
+	_ = envFile.Sync() // intentionally ignore error, as godotenv.Write does.
```

Comment:
Doesn't `godotenv.Write` return the error from `Sync()`, but not from `Close()`? reading [here](https://github.com/joho/godotenv/blob/e3b6eee84d15b0fa274078565ca46591f5e08876/godotenv.go#L159)

Reply:
You are correct that they are doing that now.  I was looking at the code we referenced (v1.4.0) inside VS Code (using the go to definition feature to look at the source in the module cache) and so I [saw something different](https://github.com/joho/godotenv/blob/v1.4.0/godotenv.go#L150-L166)

Given that upstream has moved forward, I'll move to this.  It's much nicer anyway.